libs/packages/components/src/lib/header/header.component.html

```diff
@@ -32,34 +31,93 @@
       <div class="usa-nav__secondary">
         <ul class="usa-nav__secondary-links">
           <li *ngFor="let link of model.secondaryLinks" class="usa-nav__secondary-item">
-            <a [routerLink]="[link.route]">
-              <fa-layers [fixedWidth]="true">
-                <fa-icon [icon]="[link.imageClassPrefix, link.imageClass]"></fa-icon>
-                <fa-layers-counter *ngIf="link.hasCounter" [classes]="['icon-layers-counter']"></fa-layers-counter>
-              </fa-layers>
-              {{link.text}}
-            </a>
+            <ng-container
+              [ngTemplateOutlet]="isLinkInternal(link)  ? secondaryRouteLinkTemplate :   isLinkExternal(link)? secondaryHREFLinkTemplate : secondaryEVENTLinkTemplate"
+              [ngTemplateOutletContext]="{$implicit:link}"></ng-container>
           </li>
         </ul>
       </div>
     </div>
   </nav>
 </header>
 
+<ng-template #secondaryRouteLinkTemplate let-link>
+  <a [attr.id]="link.id" [routerLink]="[link.route]" [attr.class]="link.selected ? 'usa-current' :''">
```

Comment:
For internal links, should we switch to routerLinkActive directive? I think this might work since external will never be active and event based is sort of a wildcard in the sense we don't really know what would trigger activating that link.

Reply:
So the reason I did not do that was because it is not just the link itself which indicates for the active but also the children of that link which I could not get to work with the activelink.

Reply:
Interesting, assuming the link contains the full path and it's not set to exact, then it should activate on child routes as well. 

Let's keep what you have for now, we can always debug this approach in a branch if we want to pursue it.

Reply:
There were cases where the path is under but not really under.